### PR TITLE
Fail curl command registration

### PR DESCRIPTION
**Pull request by amolpati30**
### Problem Statement
Customer was trying to register client using global registration template and while executing curl command the client was registered but it failed at one point and the echo $? showed exit status as 0.

### Solution
Giving status code 0 if passed else 1

### CODE REVIEW

**Comment by amolpati30** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_registration.py::test_negative_generate_curl_command_and_fail_registration_intentionally

**Comment by amolpati30** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_registration.py::test_negative_generate_curl_command_and_fail_registration_intentionally

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 5710
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/api/test_registration.py::test_negative_generate_curl_command_and_fail_registration_intentionally
Test Result : ================== 3 passed, 40 warnings in 803.25s (0:13:23) ==================
```

**Comment by amolpati30** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_registration.py::test_negative_generate_curl_command_and_fail_registration_intentionally

File: tests/foreman/api/test_registration.py [hunk 1]
```diff
@@ -326,3 +326,36 @@ def test_positive_host_registration_with_non_admin_user_with_setup_false(
...
+    :steps:
+        1. Generate a curl command and make the registration fail intentionally.
+        2. Check the exit code for the command.
+
+    :expectedresults: Exit code returns 1 if registration fails and returns 0 otherwise.
+
+    :BZ: 2155444
+
+    :customerscenario: true
+    """
```
Comment (shubhamsg199) [inline tests/foreman/api/test_registration.py:350]:
```suggestion
    :parametrized: yes
    """
```

File: tests/foreman/api/test_registration.py [hunk 2]
```diff
@@ -326,3 +326,36 @@ def test_positive_host_registration_with_non_admin_user_with_setup_false(
...
+    ac_key = module_target_sat.api.ActivationKey(unlimited_hosts=True).create()
```
Comment (shubhamsg199) [inline tests/foreman/api/test_registration.py:351]: You can use `module_activation_key` fixture here

**Review by shubhamsg199 — CHANGES_REQUESTED** [review]

File: tests/foreman/api/test_registration.py (hunk 2 above)
Comment (amolpati30) [reply tests/foreman/api/test_registration.py:352]: i tried with `module_activation_key` fixture but it was registering the system successfully. I wanted to fail curl intentionally so i took above code.

**Comment by amolpati30** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_registration.py::test_negative_generate_curl_command_and_fail_registration_intentionally

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 5737
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/api/test_registration.py::test_negative_generate_curl_command_and_fail_registration_intentionally
Test Result : ================== 3 passed, 52 warnings in 941.41s (0:15:41) ==================
```

File: tests/foreman/api/test_registration.py [hunk 3]
```diff
@@ -326,3 +326,37 @@ def test_positive_host_registration_with_non_admin_user_with_setup_false(
...
+    command = module_target_sat.api.RegistrationCommand(
+        organization=module_sca_manifest_org,
+        activation_keys=[ac_key.name],
+        location=module_location,
+    ).create()
+    result = rhel_contenthost.execute(command)
+    if 'The system has been registered' in result.stdout:
```
Comment (lhellebr) [inline tests/foreman/api/test_registration.py:359]: So what result do you expect? Should this pass or fail? I don't see any parametrization except for RHEL version here so only one result should be correct. This looks conceptually flawed.

**Review by lhellebr — CHANGES_REQUESTED** [review]

File: tests/foreman/api/test_registration.py (hunk 3 above)
Comment (amolpati30) [reply tests/foreman/api/test_registration.py:359]: it should fail and it is failing due to activation key. please suggest any changes are require

File: tests/foreman/api/test_registration.py [hunk 4]
```diff
@@ -326,3 +326,37 @@ def test_positive_host_registration_with_non_admin_user_with_setup_false(
     assert '# Updating packages' not in result.stdout
     # verify foreman-proxy ssh pubkey isn't present when Setup REX is false
     assert rhel_contenthost.execute('cat ~/.ssh/authorized_keys | grep foreman-proxy').status == 1
+
+@pytest.mark.rhel_ver_match('[^6]')
+def test_negative_generate_curl_command_and_fail_registration_intentionally(
```
[the remark this answers is not in the document]
Comment (amolpati30) [reply tests/foreman/api/test_registration.py]: test name changes as per suggestion.

File: tests/foreman/api/test_registration.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (amolpati30) [reply tests/foreman/api/test_registration.py]: i have make changes in the code and it only refer bash exit status code when registration get fails.

Comment (lhellebr) [reply tests/foreman/api/test_registration.py]: This new version fixes the logical flaw. Just a suggestion, you may want to `assert 'something' in result.stdout`

**Review by lhellebr — APPROVED** [review]

File: tests/foreman/api/test_registration.py [hunk 5]
```diff
@@ -326,3 +326,36 @@ def test_positive_host_registration_with_non_admin_user_with_setup_false(
...
+
+    :parametrize: yes
+    """
+    ac_key = module_target_sat.api.ActivationKey(unlimited_hosts=True).create()
+    command = module_target_sat.api.RegistrationCommand(
+        organization=module_sca_manifest_org,
+        activation_keys=[ac_key.name],
+        location=module_location,
+    ).create()
+    result = rhel_contenthost.execute(command)
```
Comment (lhellebr) [inline tests/foreman/api/test_registration.py:360]: Plase comment on why this is expected to fail.

File: tests/foreman/api/test_registration.py [hunk 6]
```diff
@@ -326,3 +326,36 @@ def test_positive_host_registration_with_non_admin_user_with_setup_false(
...
+):
+    """Verify status code, when curl command registration fail intentionally
+
+    :id: 4789e8da-6391-4ea4-aa0d-73c93220ce44
+
+    :steps:
+        1. Generate a curl command and make the registration fail intentionally.
+        2. Check the exit code for the command.
+
+    :expectedresults: Exit code returns 1 if registration fails and returns 0 otherwise.
```
Comment (lhellebr) [inline tests/foreman/api/test_registration.py:346]: Fix this to expect failure

**Review by lhellebr — CHANGES_REQUESTED** [review]

**Comment by ogajduse** [comment]
@amolpati30 can you please take a look at the failures in the CI/CQ workflow?

File: tests/foreman/api/test_registration.py [hunk 7]
```diff
@@ -326,3 +326,39 @@ def test_positive_host_registration_with_non_admin_user_with_setup_false(
...
+    :customerscenario: true
+
+    :parametrized: yes
+    """
+    ac_key_name = gen_string('alpha')
+    ac_key = module_target_sat.api.ActivationKey(name=ac_key_name).create()
+    # create RegistrationCommand with invalid activation_keys
+    command = module_target_sat.api.RegistrationCommand(
+        organization=module_sca_manifest_org,
+        activation_keys=[ac_key.name],  # activation_keys generated only with name parameter
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_registration.py:359]:
```suggestion
        activation_keys=[ac_key.name], 
```

File: tests/foreman/api/test_registration.py [hunk 8]
```diff
@@ -326,3 +326,39 @@ def test_positive_host_registration_with_non_admin_user_with_setup_false(
...
+    :expectedresults: Exit code returns 1 if registration fails.
+
+    :BZ: 2155444
+
+    :customerscenario: true
+
+    :parametrized: yes
+    """
+    ac_key_name = gen_string('alpha')
+    ac_key = module_target_sat.api.ActivationKey(name=ac_key_name).create()
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_registration.py:355]:
```suggestion
    ak = module_target_sat.api.ActivationKey(name=gen_string('alpha')).create()
```

File: tests/foreman/api/test_registration.py [hunk 9]
```diff
@@ -326,3 +326,39 @@ def test_positive_host_registration_with_non_admin_user_with_setup_false(
...
+    # create RegistrationCommand with invalid activation_keys
+    command = module_target_sat.api.RegistrationCommand(
+        organization=module_sca_manifest_org,
+        activation_keys=[ac_key.name],  # activation_keys generated only with name parameter
+        location=module_location,
+    ).create()
+    result = rhel_contenthost.execute(command)
+    assert result.status == 1
+    assert "Couldn't find activation key" in result.stderr
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_registration.py:364]:
```suggestion
    assert 'Couldn't find activation key' in result.stderr
```

File: tests/foreman/api/test_registration.py [hunk 10]
```diff
@@ -326,3 +326,39 @@ def test_positive_host_registration_with_non_admin_user_with_setup_false(
...
+
+    :BZ: 2155444
+
+    :customerscenario: true
+
+    :parametrized: yes
+    """
+    ac_key_name = gen_string('alpha')
+    ac_key = module_target_sat.api.ActivationKey(name=ac_key_name).create()
+    # create RegistrationCommand with invalid activation_keys
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_registration.py:356]: Could you correct this comment as per @lhellebr's suggestion?

File: tests/foreman/api/test_registration.py (hunk 5 above)
Comment (Gauravtalreja1) [reply tests/foreman/api/test_registration.py:361]: @amolpati30 Could you check and update this?

**Review by ogajduse — APPROVED** [review]

**Review by Gauravtalreja1 — APPROVED** [review]

**Pull request merged**